### DISCUSSION TITLE
Kata 4 (¿Python sabe matematicas?)

### DISCUSSION THREAD

**Discussion by SantQuintana** [LaunchX-InnovaccionVirtual/CursoIntroPython#100] [Q&A] [answered]
Alguien sabe por que 

x = .00162 * 1000
print(x)

imprime de salida 
>>> 1.6199999999999999

Pero matematicamente debe de dar "1.62" esto no es lo que me devuelve Python.
Lo curioso es que si multiplico *10,000, si me da el resultado correcto (16.2 de resultado).
Alguien sabe por que se da esto y si tiene una solución?

**Comment by Dfredude** [comment, marked as answer]
- Que buena pregunta! Esto tiene que ver con la triste realidad de que las computadoras no saben manejar decimales! Asi como lo escuchaste. Es mas, algo tan sencillo como esto:
```python
x = 0.1+0.2
print(x)
```
Te va dar esto:
![image](https://user-images.githubusercontent.com/72330887/153520836-06ebe738-f124-4ff9-a59f-0bb3a3b8a0b6.png)
Jajaja, la maquina no sabe que la respuesta correcta es 0.3, la mayoria de los lenguajes de programacion (si no es que todos) caen al mismo error, ya que todos se basan en la misma arquitectura computacional. Java, c++, y demas te daran el mismo resultado equivocado. 

El tema tiene que ver con como funcionan los **_floating-point numbers_** lo cual queda fuera de esta discusion, pero si quieres saber mas te recomiendo investigar. Una mejor explicacion: https://www.youtube.com/watch?v=PZRI1IfStY0.

- Lo importante, es que hay librerias matematicas (math para Python, por ejemplo) que redondean los numeros, solucionando este tipo de errores matematicos.

### Espero haberte ayudado!

**Comment by MichelleEspinoza** [comment]
Podrías aplicar una función de redondeo e indicar cuantos decimales necesitas por ejemplo:
#Biblioteca Math
from math import ceil, floor

x = .00162 * 1000
print(x)
print(round(x, 2))

**Comment by doguedogue** [comment]
es la precisión únicamente, como el número π (pi) 3.141592...., yo normalmente digo 3.1416

```
x = .00162 * 1000
print(x)
print(round(.00162 * 1000, 2))
```
![image](https://user-images.githubusercontent.com/23409026/153521830-6691ae6e-b924-42ab-80d4-f23b21a21a55.png)
